rules-engine/tests/rules_engine/test_examples.py

```diff
@@ -14,18 +18,97 @@
 INPUT_DATA = next(os.walk(ROOT_DIR))[1]
 
 
-def load_example(folder: str) -> dict[str, Any]:
+def validate_fuel_type(value):
+    try:
+        return engine.FuelType[value]
+    except KeyError as e:
+        raise ValueError(
+            f"Error validating fuel type {e}. Valid choices are: {[x.name for x in engine.FuelType]}"
+        )
+
+
+class Summary(BaseModel):
+    local_weather_station: str
+    design_temperature_override: Optional[float]
+    living_area: int
+    fuel_type: Annotated[engine.FuelType, BeforeValidator(validate_fuel_type)]
+    heating_system_efficiency: float
+    other_fuel_usage: Optional[float]
+    other_fuel_usage_override: Optional[float]
+    thermostat_set_point: float
+    setback_temp: float
+    setback_hours_per_day: float
+    estimated_balance_point: float
+    balance_point_sensitivity: float
+    average_indoor_temperature: float
+    difference_between_ti_and_tbp: float
+    design_temperature: float
+    whole_home_ua: int
+    standard_deviation_of_ua: float
+    avg_heat_load: int
+    max_heat_load: int
+
+
+def validate_datetime(value):
+    return datetime.strptime(value, "%m/%d/%Y").date()
```

Comment:
Should we check for error here?

Reply:
`validate` might be a misnomer.  It's really more about parsing, and validation is a side effect, in that it will error out if the format is wrong.  For example:

```py
>>> from datetime import datetime
>>> datetime.strptime("13/13/2013", "%m/%d/%Y").date()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Alan Pinkert\.pyenv\pyenv-win\versions\3.9.6\lib\_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
  File "C:\Users\Alan Pinkert\.pyenv\pyenv-win\versions\3.9.6\lib\_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data '13/13/2013' does not match format '%m/%d/%Y'
```